### DISCUSSION TITLE
Which config settings to use for new retention by available storage feature [0.12.0-beta1]

### DISCUSSION THREAD

**Discussion by thinkloop** [blakeblackshear/frigate#4925] [Ask A Question]
What kind of settings should we use in the config.yml under recordings and events with regards to the new "available storage" feature? The comments on the [new docs page](https://deploy-preview-4055--frigate-docs.netlify.app/configuration/) seem to be outdated:

![image](https://user-images.githubusercontent.com/118814/210876392-d7563ff2-9adc-417a-8c36-7d6eb450346f.png)

Does this feature affect "events" at all? Is it possible to keep events forever until space is needed?

Thank you.

**Comment by NickM-27** [comment]
> What kind of settings should we use in the config.yml under recordings and events with regards to the new "available storage" feature?

I don't see any reason to change settings, it's simply a matter of what you want to record. If you were restricting the days due to fear of crash then it can be increased now without worry.

> The comments on the [new docs page](https://deploy-preview-4055--frigate-docs.netlify.app/configuration/) seem to be outdated:

which comments?

> Does this feature affect "events" at all? Is it possible to keep events forever until space is needed?

it affects recordings which are a part of events, but snapshots will always be deleted based on the days and this feature does not consider / delete snapshots for storage.

**Reply by thinkloop** [reply]
> which comments?

I guess I thought the `record: retain: days:` would be deprecated but I see what you mean now. As an aside, it might make sense to mention the new feature in the comments to help make decisions.

In general, I have a drive dedicated to frigate, ideally I would like it to manage itself by default, using all its space without my having to think about it. If I put 1 million days to retain events and 90 days for record, would that work?

**Reply by NickM-27** [reply]
> it might make sense to mention the new feature in the comments to help make decisions.

yeah, will do

> In general, I have a drive dedicated to frigate, ideally I would like it to manage itself by default, using all its space without my having to think about it.

That PR hasn't been closed due to not all the features being met. We know there will be more requests for options to configure like a certain % storage remaining or something which should come in the future. 

>  If I put 1 million days to retain events and 90 days for record, would that work?

I mean, it wouldn't crash, I don't see a reason to do that though as opposed to just setting 90 days for recording, events, and snapshots and just let it do its thing.

**Comment by thinkloop** [comment]
> I mean, it wouldn't crash, I don't see a reason to do that though as opposed to just setting 90 days for recording, events, and snapshots and just let it do its thing.

Maybe I am missing something here, but my drive can store much more than 90 days of events. I currently have it at 730 days (2 years) with record at 90 days. But events are hard to predict, I would have to sample each camera and multiply it out, then keep monitoring the environment to see if something structural changed increasing or decreasing the average number of events. 

I see "recordings" as work files, and care only to store as many events as my drive will allow. If I do my 1 million days thing will it continue to record more than 2 years if there is space FIFO style without crashing and with recordings set to 90 days?

**Reply by NickM-27** [reply]
>  If I do my 1 million days thing will it continue to record more than 2 years if there is space FIFO style?

It will continue to record until there is < 1 hour of recording space left at which point it will remove 1 hour to add to a total of 2 hours space, then continue so on and so forth

**Reply by thinkloop** [reply]
Actually I probably am missing something - why is recordings exposed at all? Why not just have one interface to select what you want to record and for how long?

**Reply by NickM-27** [reply]
I don't follow your question, are you asking why there are different settings for events and 24/7 recording?

**Reply by thinkloop** [reply]
I didn't realize recording was for 24/7, I thought it was to power the precapture/postcapture for events. So to have 24/7 the retain mode would have to be "all"? If the modes for both record and events were "active_objects" does that mean record and events would be identical? Would that be an unnecessary combination?

**Reply by NickM-27** [reply]
I'd highly suggest you read the docs https://docs.frigate.video/configuration/record#what-do-the-different-retain-modes-mean

`record -> retain -> days` is 24/7 recording always. If the retain mode is all then it will keep **all** 24/7 segments. If the retain mode is **motion** then it will retain every single 24/7 segment that has motion. If the retain mode is **active_objects** then it will keep every single 24/7 segment that has an active object in it.

The `record -> events -> retain -> default` is similar except only applies to recording segments that are relevant for events.

**Reply by thinkloop** [reply]
Thank you. So it is true that if both record and events are set to "active_objects" that they would be identical? If "events" is set to "all" then what is the definition of an event to trigger it?

**Reply by NickM-27** [reply]
> So it is true that if both record and events are set to "active_objects" that they would be identical?

It depends on your config. If you have events setup so any object that is detected saves an event then yes I suppose, but if you have required_zones then it would be entirely different 

So lets's say for events you have required_zones for your driveway. The 24/7 recording with active_objects would still save this recording even if the object never entered your driveway. Where the events active_objects only applies to events (which require an object to enter the driveway zone).

> And last one, if "events" is set to "all" then what is the definition of an event to trigger it?

The same as it is no matter what. These settings only apply to how recordings are handled and have no affect on the logic behind what is an event and what isn't

**Reply by thinkloop** [reply]
> The same as it is no matter what. These settings only apply to how recordings are handled and have no affect on the logic behind what is an event and what isn't

I thought events were either motion or active_objects as defined under events,I tried searching the docs for what the definition of an event is but couldn't find it. The page you linked gives the example of a car driving up to trigger a full 4 hour recording if "all" is selected, was the car driving up triggered due to "motion"?

**Reply by NickM-27** [reply]
> I tried searching the docs for what the definition of an event is.

events are object detections, the object is tracked the entire time it is visible and each object has their own event. Depending on your config the event may not be saved until the object enters specified zones. The event for that object goes on as long as that object is visible.

> The page you linked gives the example of a car driving up to trigger a full 4 hour recording if "all" is selected, was the car driving up "motion"?

No, the car driving up is an object detected. Events are only based on objects, there are no events for motion. So the recording retention settings are for setting which recording segments to keep relative to when that object was visible. 

The `record -> events -> retain -> mode` option asks you if you want to keep:
`all` 10 second segments when the object was visible
segments when the object was visible and there was `motion`
or segments when the object was visible and any object in the frame was considered active.

**Reply by thinkloop** [reply]
Ok, thank you 👍
